src/lib/Libifl/auth.c

```diff
@@ -1089,6 +1091,50 @@ auth_exec_socket(int sock, unsigned short port, char *auth_method, char* jobid)
 			return INTERACTIVE_AUTH_RETRY;
 		}
 		free_auth_config(auth_config);
+	} else if ((strcmp(auth_method, AUTH_GSS_NAME) == 0)) {
+		char encrypt_method[MAXAUTHNAME + 1] = "GSS";
```

Comment:
wait, so we always have encrytpion when auth_method is GSS? Can we separate the two? i.e. we might want to have GSS authentication, without encryption.

Reply:
I am not very happy with the possibility of using GSS without encryption in general. The reason is that the credentials/krb ticket is channeled through the channel `server<->mom`.

I must admit, here it is something else... We do not transfer credentials `qsub<->mom`. So, I added the env `PBS_INTERACTIVE_ENCRYPT_METHOD` and tested it:
```
torque2.grid.cesnet.cz$ PBS_INTERACTIVE_AUTH_METHOD=GSS PBS_INTERACTIVE_ENCRYPT_METHOD= qsub -I 
qsub: waiting for job 117523.torque1.grid.cesnet.cz to start
qsub: job 117523.torque1.grid.cesnet.cz ready

torque1.grid.cesnet.cz$ exit
logout

qsub: job 117523.torque1.grid.cesnet.cz completed
torque2.grid.cesnet.cz$ 
```
I verified that `tcpdump` shows unencrypted data in this case.

Reply:
The job starts just fine. But no encryption is done and the communication is unencrypted.

The same happens with `PBS_AUTH_METHOD=resvport PBS_ENCRYPT_METHOD=GSS qstat`.

I can test that the env `*ENCRYPT_METHOD=gss` requires `*AUTH_METHOD=gss`. Or somthing more general?

Reply:
Hm, so if the communication is not encrypted, then we should better return an error and not let the job to start.